src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -140,8 +137,16 @@ public Map<Integer, Float> searchLeaf(LeafReaderContext context, int k) throws I
          * This improves the recall.
          */
         Map<Integer, Float> docIdsToScoreMap;
+        final ExactSearcher.ExactSearcherContext exactSearcherContext = ExactSearcher.ExactSearcherContext.builder()
+            .k(k)
+            .isParentHits(true)
```

Comment:
nit: Is this correct? if yes I think it needs a comment why is it kept true

Reply:
These were the defaults which was present. I can add a comment here.

Reply:
NVM. the comment is here: https://github.com/opensearch-project/k-NN/pull/2076/files#diff-68bee8536a6d51809bc0408f057e47c6f2c1921a0eef56488f2b54cc1986a85bR177-R182

Reply:
will add some comment here too.